glean-core/android/src/main/java/mozilla/telemetry/glean/Glean.kt

```diff
@@ -789,10 +795,17 @@ open class GleanInternalAPI internal constructor () {
      */
     @Synchronized
     internal fun registerPingType(pingType: PingTypeBase) {
-        if (this.isInitialized()) {
-            LibGleanFFI.INSTANCE.glean_register_ping_type(
-                pingType.handle
-            )
+        // If this happens after Glean.initialize is called (and returns),
+        // we dispatch ping registration on the thread pool.
+        // Registering a ping should not block the application.
+        // Submission itself is also dispatched, so it will always come after the registration.
+        if (this.initFinished) {
```

Comment:
Mh, this is interesting. [initFinished](https://github.com/mozilla/glean/blob/fa2810fdeeea08f3f827c71ed5f37f0aef240c1e/glean-core/android/src/main/java/mozilla/telemetry/glean/Glean.kt#L262) is set outside of the dispatched init, which means this is automatically set to `true` whenever `Glean.Initialize` is called, regardless of the fact that the dispatched init has executed or not. This might be another bug, but leads to two questions:

- was that intentional?
- is the `initFinished` really a misnamed variable that should be `initDispatched`?
- If that was intentional, then this check is not really helping too much. as we really want to make sure this is exclusively happening [after this](https://github.com/mozilla/glean/blob/fa2810fdeeea08f3f827c71ed5f37f0aef240c1e/glean-core/android/src/main/java/mozilla/telemetry/glean/Glean.kt#L203-L211) , otherwise we might loose some ping registration.

Reply:
>     * was that intentional?

Yes, we only need to track that `initialize` was called.

>     * is the `initFinished` really a misnamed variable that should be `initDispatched`?

Happy to bikeshed on the name, yes.

>     * If that was intentional, then this check is not really helping too much. as we really want to make sure this is exclusively happening [after this](https://github.com/mozilla/glean/blob/fa2810fdeeea08f3f827c71ed5f37f0aef240c1e/glean-core/android/src/main/java/mozilla/telemetry/glean/Glean.kt#L203-L211) , otherwise we might loose some ping registration.

It is helping. We put it on the dispatch queue, which is not worked on until `Dispatchers.API.flushQueuedInitialTasks()` is called late in the dispatched initialize part.

Reply:
Got it. I'm now convinced of this :) Thanks for bearing with me.